extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -344,6 +435,38 @@ export abstract class AzureAuth implements vscode.Disposable {
 	}
 
 
+	public async getTenantsMsal(token: string): Promise<Tenant[]> {
+		interface TenantResponse { // https://docs.microsoft.com/en-us/rest/api/resources/tenants/list
+			id: string
+			tenantId: string
+			displayName?: string
+			tenantCategory?: string
+		}
+		const tenantUri = url.resolve(this.metadata.settings.armResource.endpoint, 'tenants?api-version=2019-11-01');
+		try {
+			Logger.verbose('Fetching tenants', tenantUri);
+			const tenantResponse = await this.makeGetRequest(tenantUri, token);
+			const tenants: Tenant[] = tenantResponse.data.value.map((tenantInfo: TenantResponse) => {
+				Logger.verbose(`Tenant: ${tenantInfo.displayName}`);
```

Comment:
Instead of having separate log messages for each tenant, which may be separated from the other ones (if other messages are coming in at the same time) I would do a single message after the map which prints all the display names of the tenants retrieved.